src/providers/whatsapp/whatsapp.service.ts

```diff
@@ -136,4 +174,7 @@ export class WhatsappService {
     const formattedNumber = `${number}@c.us`;
     return formattedNumber;
   }
+  private isLogged() {
+    if (!this.client) return true;
+  }
```

Comment:
```suggestion
  private isLogged() {
    if (!this.client) throw new Error("Whatsapp desconectado")
  }
```
percebe que tu chama esse método em todas as outras funções e faz um if para depois retornar um  ```{ error: 'Whatsapp desconectado' }```.  

O ideal é tu colocar essa verificação direto nesse método novo e caso o whatsapp esteja desconectado tu deve parar a request lançando um erro, assim como fiz nessa sugestão.

Sempre que tu precisar lançar um erro, usar a classe Error do javascript ouuuu as classes de exception do nestjs https://docs.nestjs.com/exception-filters

Reply:
eu tentei utilizar esse throw new Error, mas ele não retorna na response, aparece internal server error, do jeito que eu tava fazendo aparecia no response, era pra aparecer com o throw new Error?

Reply:
é pra aparecer, fiz um teste aqui e apareceu

Reply:
posso te mandar print depois

Reply:
puts

Reply:
por aqui ele ta lançando a exceção no terminal só